mangum/scope.py

```diff
@@ -0,0 +1,55 @@
+from dataclasses import dataclass, field
+from typing import List, Tuple, Optional, Dict, Any, Union, TYPE_CHECKING
+
+from mangum.types import ScopeDict
+
+if TYPE_CHECKING:  # pragma: no cover
+    from awslambdaric.lambda_context import LambdaContext
+
+
+@dataclass
+class Scope:
```

Comment:
Alternatively, maybe `Request` instead of `Event` makes more sense here.

Reply:
I'd like to leave this one scope _solely_ because that's what the ASGI folks call it: https://asgi.readthedocs.io/en/latest/specs/www.html#http-connection-scope

Similarly Event is what AWS calls all of these different lambda triggers. Keeping with existing conventions.

Reply:
I'll move that as_dict() call till later though, might keep things more structured until the very last possible moment

Reply:
This should be changed from `Scope` to something else because it actually encompasses more than just the `Scope` in the class definition - the `ScopeDict` itself is would be more consistent with the other types (e.g. we don't have `ReceiveCallback`). So rather than `Event` or `Scope` it seems `Request` would be most appropriate because it contains both the lambda event information and the ASGI request information that ultimately becomes the scope later.